### PR TITLE
Rework Config

### PR DESCRIPTION
**Pull request by wjp**
This PR reworks the Config system. It moves directly reading from the XML-based Config to a new `ConfigReader` class, introducing an implementation-independent interface. This also massively reduces code duplication and simplifies object initialization functions.

Writing Configs from the geometry classes now goes via a tiny `ConfigWriter` class.

The only code now still touching XMLNodes directly is in the matlab/python interfaces.

This is a first step towards possibly removing the intermediate XML layer, and directly accessing Python dicts or Matlab structs.

### CODE REVIEW

**Comment by Obi-Wan** [comment]
This seems like a very cool change. Is it going have an impact on the projector initialization speed?

**Comment by wjp** [comment]
In the current stage it's mainly cleaning up the code, but if/when we do replace the XML backend by a direct python dict backend (or something similar), that should make initialization a lot faster, yes.

**Comment by Obi-Wan** [comment]
Ok, that would be great for a couple of scenarios where I need to create many projection geometries.

Anyway, this is great work! (I mean, your work is always great... I'm just happy that it is focused on astra right now :) )

File: src/SartAlgorithm.cpp [hunk 1]
```diff
@@ -146,8 +145,8 @@ bool CSartAlgorithm::initialize(const Config& _cfg)
 		}
 		for (int i = 0; i < projectionCount-1; i++) {
 			int k = (rand() % (projectionCount - i));
-			int t = projectionOrder[i];
-			m_piProjectionOrder[i] = projectionOrder[i + k];
+			int t = m_piProjectionOrder[i];
+			m_piProjectionOrder[i] = m_piProjectionOrder[i + k];
```
Comment (wjp) [inline src/SartAlgorithm.cpp:149]: Thanks for catching that! It looks like I was confused by near similarities with `CudaSartAlgorithm` here...

**Pull request merged**